src/sparseml/transformers/sparsification/trainer.py

```diff
@@ -365,6 +386,12 @@ def save_model(self, output_dir: Optional[str] = None, _internal_call=True):
         recipe_path = os.path.join(
             output_dir, RECIPE_TEMPLATE.format(f"_{index:02d}" if index > 0 else "")
         )
+        checkpoint_path = os.path.join(self.args.output_dir, RECIPE_NAME)
+
+        if os.path.isfile(checkpoint_path):
+            self.manager = ScheduledModifierManager.compose_staged(
```

Comment:
important - we do not want to overwrite the actual manager here with the composed recipe (this should always be just the training recipe), should just be a temporary composed_manager variable

Reply:
Valid comment, but I propose this gets done in the parallel PR (#664) since the final location of this functionality will be determined there.